src/credential/mod.rs

```diff
@@ -1,45 +1,159 @@
+}
+
+#[uniffi::export]
+impl Credential {
+    /// Create a new credential.
+    #[uniffi::constructor]
+    pub fn new(
+        id: Uuid,
+        format: CredentialFormat,
+        r#type: CredentialType,
+        payload: Vec<u8>,
+        key_alias: Option<KeyAlias>,
+    ) -> Arc<Self> {
+        Arc::new(Self {
+            id,
+            format,
+            r#type,
+            payload,
+            key_alias,
+        })
+    }
+
+    /// Return the ID of the credential.
+    pub fn id(&self) -> Uuid {
+        self.id
+    }
+
+    /// Return the format of the credential.
+    pub fn format(&self) -> CredentialFormat {
+        self.format.clone()
+    }
+
+    /// Return the type of the credential.
+    pub fn r#type(&self) -> CredentialType {
+        self.r#type.clone()
+    }
+
+    /// Return the raw payload of the credential.
+    pub fn payload(&self) -> Vec<u8> {
+        self.payload.clone()
+    }
+}
+
+// Internal helper methods.
+impl Credential {
+    /// Convert the parsed credential into a specialized JSON credential.
+    pub fn try_into_parsed(&self) -> Result<Arc<ParsedCredential>, CredentialDecodingError> {
+        self.to_owned().try_into()
+    }
+
+    /// Returns the credential as a `Mdoc` credential.
+    pub fn try_into_mdoc(&self) -> Result<Arc<Mdoc>, CredentialDecodingError> {
+        self.to_owned()
+            .try_into()
+            .map_err(|e: MdocInitError| CredentialDecodingError::MsoMdoc(e.to_string()))
+    }
+
+    /// Returns the credential as a `JwtVc` credential.
+    pub fn try_into_jwt_vc(&self) -> Result<Arc<JwtVc>, CredentialDecodingError> {
+        self.to_owned()
+            .try_into()
+            .map_err(|e: JwtVcInitError| CredentialDecodingError::JwtVc(e.to_string()))
+    }
+
+    /// Return the credential as an AnyJsonCredential type.
+    pub fn try_into_any_json_credential(
+        &self,
+    ) -> Result<AnyJsonCredential, CredentialDecodingError> {
+        self.to_owned().try_into()
+    }
+}
+
+impl TryFrom<Credential> for AnyJsonCredential {
+    type Error = CredentialDecodingError;
+
+    fn try_from(value: Credential) -> Result<Self, Self::Error> {
+        match value.format {
+            // NOTE: Re-using the Arc<JwtVc> type for SdJwtJson format
+            // for converting into the AnyJsonCredential type.
+            //
+            // There may be a better solution for this.
+            CredentialFormat::JwtVcJson => {
+                let jwt_vc: Arc<JwtVc> = value
+                    .try_into()
+                    .map_err(|e: JwtVcInitError| CredentialDecodingError::JwtVc(e.to_string()))?;
+                Ok(jwt_vc.credential())
+            }
+            CredentialFormat::SdJwt => {
+                let sd_jwt: Arc<SdJwt> = value
+                    .try_into()
+                    .map_err(|e: SdJwtError| CredentialDecodingError::SdJwt(e.to_string()))?;
+
+                Ok(sd_jwt.credential())
+            }
```

Comment:
As above, I think we should always work through ParsedCredential for consistency.

Reply:
This method has now been removed completely.